akka-actor-tests/src/test/scala/akka/actor/ActorLifeCycleSpec.scala

```diff
@@ -153,4 +154,33 @@ class ActorLifeCycleSpec
     }
   }
 
+  "have a non null context after termination" in {
+    class StopBeforeFutureFinishes(val latch: CountDownLatch) extends Actor {
+      import context.dispatcher
+      import akka.pattern._
+
+      override def receive: Receive = {
+        case "ping" =>
+          val replyTo = sender()
+
+          context.stop(self)
+
+          Future {
+            latch.await()
+            "po"
+          }.flatMap(x => Future { x + "ng" }).recover { case _: NullPointerException => "npe" }.pipeTo(replyTo)
```

Comment:
Isn't it supposed to touch context in the future then? Here we only use the sender that was captured before the actor stopped.

Reply:
the context is implicitly referenced here through `context.dispatcher` as an executor context for futures.

Reply:
Ah 👍